jetty-core/jetty-http2/jetty-http2-server/src/main/java/org/eclipse/jetty/http2/server/internal/HttpStreamOverHTTP2.java

```diff
@@ -84,6 +87,14 @@ public Runnable onRequest(HeadersFrame frame)
 
             Runnable handler = _httpChannel.onRequest(_requestMetaData);
 
+            ComplianceViolation.Listener listener = Server.newComplianceViolationListener(_httpChannel.getConnectionMetaData().getConnector());
```

Comment:
This implies that we are making new listener instances even if none of them are capturing.

Reply:
The signature is "new" as that's the contract on use of the method.
If it returns the same instance on call is just an implementation detail.
We don't want a "get" semantic as that implies it can be called repeatedly to access the same one (which is not true).